### PR TITLE
Add optional override of the namespace tiller will be launched into

### PR DESCRIPTION
**Pull request by karcaw**
Signed-off-by: Evan Felix <evan.felix@pnnl.gov>

@krancour 

**What this PR does / why we need it**:
This adds the ability to specify the namespace that the chart will deploy into, overriding anything specified on the helm command line.  This is needed when we use it as a subchart to deploy namespaces that are different from the namespace that is used on the command line.  A simple example would be if I used this as a subchart twice, and needed to specify two different namespaces.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by jonhines** [comment]
+1 for this change

**Comment by krancour** [comment]
@karcaw sorry for the delay in review.

I understand your use case, but _initially_ had a concern over whether this was safe. I had to experiment a bit...

Ordinarily, I would say that a chart absolutely should not do this, because once a namespace is counted as a "Helm-managed resource," deleting the chart will delete the namespace and that will potentially take _other_ things with it-- for instance, other charts or loose manifests that have been deployed into the same namespace.

This chart feels like it should be an exception to that _because_ it's all about sane management of a namespace. So, my principled concern expressed above is moot. On to the subject of safety... and some experiments...

Assume the following:

* You (as a cluster admin) have deployed this chart to create managed namespace `foo`
* You (as a cluster admin) have used the `kube-system` tiller to deploy other bits of infra, middleware, etc. into the the `foo` namespace. We'll call this `bar`. You've used the `kube-system` tiller for this, btw, because that's the "admin's tiller" and it can do loads of things that the tiller in the `foo` namespace (the developer's tiller) probably cannot do.
* You (again, as the cluster admin) delete the `foo` release.
    * It takes the `foo` namespace with it.

What happens to the `bar` release? I checked-- it gets orphaned. All its resources will be gone, but `helm ls` will still show it. How much of a problem is this? Not much, as it turns out. `helm delete bar --purge` still works just fine.

There's one other scenario to consider.

* You (as a cluster admin) have deployed this chart to create managed namespace `foo`
* A developer has used the `foo` tiller to deploy some software into this namespace. We'll call this `bar`.
* You (as the cluster admin) delete the `foo` release.
    * It takes the `foo` namespace with it.

In this case the `ConfigMap`(s) that hold the metadata for the `bar` release are all confined to the `foo` namespace and deleting it takes all the developer's releases with it. Nothing at all gets orphaned. It's a nice tidy delete.

Bottom line-- this feels ok to me, but it took some experimentation to convince myself of that.

One change it required, however. The `README.md` states explicitly that deleting a release of this chart _does not_ delete the namespace itself. This PR makes that behavior conditional. The documentation really needs to be updated to reflect that.

Otherwise, lgtm. Update the docs and we can merge this.

**Comment by krancour** [comment]
Thanks @karcaw!

/approve
/lgtm

**Comment by kfox1111** [comment]
@krancour can we get an ok-to-test?
